app/src/pages/admin/allUsersPage/allUsersGrid/projectsAndRolesColumn/projectsAndRolesColumn.jsx

```diff
@@ -1,31 +1,261 @@
+@track()
+@injectIntl
 export class ProjectsAndRolesColumn extends Component {
   static propTypes = {
+    intl: intlShape.isRequired,
     className: PropTypes.string.isRequired,
     value: PropTypes.object,
+    tracking: PropTypes.shape({
+      trackEvent: PropTypes.func,
+      getTrackingData: PropTypes.func,
+    }).isRequired,
+    showNotification: PropTypes.func.isRequired,
+    showModalAction: PropTypes.func.isRequired,
+    fetchAllUsersAction: PropTypes.func,
+    toggleUserRoleForm: PropTypes.func,
   };
   static defaultProps = {
     value: {},
+    fetchAllUsersAction: () => {},
+    toggleUserRoleForm: () => {},
   };
   state = {
-    expanded: false,
+    assignRole: false,
+    isMobileView: false,
+  };
+
+  componentDidMount() {
+    this.match = window.matchMedia(SCREEN_XS_MAX);
+    this.match.addListener(this.setMobileView);
+    this.setMobileView(this.match);
+  }
+
+  componentWillUnmount() {
+    if (!this.match) {
+      return;
+    }
+    this.match.removeListener(this.setMobileView);
+  }
+  onChange = (project, role) => {
+    const {
+      intl,
+      value: { userId },
+    } = this.props;
+    fetch(URLS.project(project), {
+      method: 'put',
+      data: { users: { [userId]: role } },
+    })
+      .then(() => {
+        this.props.showNotification({
+          message: intl.formatMessage(messages.updateUserRole, {
+            user: userId,
+          }),
+          type: NOTIFICATION_TYPES.SUCCESS,
+        });
+        this.props.fetchAllUsersAction();
+      })
+      .catch((err) => {
+        this.props.showNotification({
+          message: err.message,
+          type: NOTIFICATION_TYPES.ERROR,
+        });
+      });
+  };
+  onAssignProjectRole = (project, role) => {
+    const {
+      intl,
+      value: { userId },
+    } = this.props;
+    fetch(URLS.userInviteInternal(project), {
+      method: 'put',
+      data: { userNames: { [userId]: role } },
+    })
+      .then(() => {
+        this.props.showNotification({
+          message: intl.formatMessage(messages.updateUserRole, {
+            user: userId,
+          }),
+          type: NOTIFICATION_TYPES.SUCCESS,
+        });
+        this.props.fetchAllUsersAction();
+      })
+      .catch((err) => {
+        this.props.showNotification({
+          message: err.message,
+          type: NOTIFICATION_TYPES.ERROR,
+        });
+      });
+  };
+  onDelete = (project) => {
+    const {
+      value: { userId },
+    } = this.props;
+    this.showUnassignModal(project, userId);
+  };
+  setMobileView = (media) =>
+    media.matches !== this.state.isMobileView &&
+    this.setState({
+      isMobileView: media.matches,
+    });
+  getProjectsList = () => {
+    const {
+      value: { assignedProjects = {} },
+    } = this.props;
+    return Object.keys(assignedProjects);
+  };
+  getVisibleProjects = () => this.getProjectsList().slice(0, 2);
```

Comment:
why there are only two first projects are visible?

Reply:
That's business requirements